src/Component/Provider/ArrayGridProvider.php

```diff
@@ -37,22 +44,20 @@ public function get(string $code): Grid
         }
 
         $gridConfiguration = $this->gridConfigurations[$code];
+        $parentGridCode = $gridConfiguration['extends'] ?? null;
+
+        if (null !== $parentGridCode) {
+            $parentGridConfiguration = $this->gridConfigurations[$gridConfiguration['extends']] ?? null;
 
-        if (isset($gridConfiguration['extends'], $this->gridConfigurations[$gridConfiguration['extends']])) {
-            $gridConfiguration = $this->extend($gridConfiguration, $this->gridConfigurations[$gridConfiguration['extends']]);
+            Assert::notNull($parentGridConfiguration, sprintf('Parent grid with code "%s" does not exists.', $gridConfiguration['extends']));
```

Comment:
This might break someone's code... it needs a mention in the UPGRADE file.

Reply:
done